packages/@tinacms/cli/src/server/models/media.ts

```diff
@@ -92,11 +94,21 @@ export class MediaModel {
         }
       })
 
-      const files = await Promise.all(filesProm)
+      const offset = Number(args.cursor) || 0
+      const limit = Number(args.limit) || 20
+
+      const allItems = await Promise.all(filesProm)
+      const files = allItems.filter((x) => x.isFile)
+      const directories = allItems.filter((x) => !x.isFile).map((x) => x.src)
+
+      const limitFiles = files.slice(offset, offset + limit)
```

Comment:
@Enigmatical in the tina-cloud implementation, for paging purposes, there is no distinction between files and directories. Here it looks like if you specify a limit of 20, and have say 25 directories, you will end up with more than 20 items in the list

Reply:
Ah, good catch.  I didn't realize directories counted against the "page" limit.  Easy to solve.